zip-0315.rst

```diff
@@ -1,9 +1,620 @@
+
+SHOULD try to create fully shielded transactions where possible.
+
+Open question: Are we going to describe the policy that zcashd uses?
+
+
+* Numbers of JoinSplits, Spends, Outputs, and Actions
+
+  * This could correlate with other information under certain circumstances.
+    For example in a “dusting attack”, the adversary sends a victim lots of
+    small-valued notes (or notes with strategically chosen values),
+    increasing the probability that the victim’s transactions will have a
+    larger number of spends than other transactions (or an identifiable
+    number of spends). There are note management strategies that can mitigate
+    this, but they have not been implemented yet.
+
+* Which spends of given notes are repeated across transactions
+
+  * This may happen because a previous transaction expired and the user is
+    trying to spend some of the same notes.
+
+* Whether the transaction is coinbase
+
+* For coinbase transactions, the amounts / destination addresses / memos of
+  shielded outputs
+
+* Orchard flags (enableSpends/enableOutputs)
+
+  * Under normal circumstances these only depend on whether the transaction
+    is coinbase.
+
+
+Linkability of transactions or addresses
+''''''''''''''''''''''''''''''''''''''''
+
+Motivation
+~~~~~~~~~~
+
+We want to support creating unlinkable addresses, in order that a user can
+give different addresses to different counterparties, in such a way that the
+counterparties (even if they collude) cannot tell that the addresses were
+provided by the same or distinct users.
+
+If multiple UTXOs are received at the same transparent address, it is safe
+to shield them all in the same transaction, because that is not leaking
+additional information.
+
+UTXOs received on different transparent receivers SHOULD NOT be shielded
+in the same transaction. Ideally, when they are shielded in separate
+transactions, this should be done in such a way that the timing of those
+transactions is not linkable.
+
+When spending transparent UTXOs, they should only be sent to an internal
+shielded receiver belonging to the wallet.
+
+A wallet MUST NOT send funds to a transparent address unless all of the
+source funds come from shielded pools, and this SHOULD be a single shielded
+pool.
+
+We want to minimize pool crossing.
+
+Anchor selection
+''''''''''''''''
+
+A wallet SHOULD choose an anchor a number of blocks back from the head of the
+chain equal to the trusted confirmation depth. That is, if the current block
+is at height H, the anchor SHOULD reflect the final treestate of the block at
+height H-3.
+
+
+Rationale for anchor selection
+''''''''''''''''''''''''''''''
+
+* If the chain rolls back past the block at which the anchor is chosen, then
+  the anchor and the transaction will be invalidated. This is undesirable
+  both for reliability, and because the nullifiers of spent notes will have
+  been revealed, linking this transaction to any future transactions that
+  spend those notes.
```

Comment:
```suggestion
* If the chain rolls back past the block at which the anchor is chosen, then
  the anchor and the transaction will be invalidated. This is undesirable
  both for reliability, and because the nullifiers of spent TXOs will have
  been revealed, linking this transaction to any future transactions that
  spend those TXOs.
```

Reply:
This only applies to notes; transparent TXOs don't have nullifiers.